pkg/daemon/cleanup/cleanup.go

```diff
@@ -57,13 +58,24 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 		klog.Exitf("could not create service manager: %s", err.Error())
 	}
 	defer svcMgr.Disconnect()
-	services, err := getServicesToRemove(ctx, directClient, node, configMapNamespace)
+	latestCMData, versionCMData, err := getLatestAndVersionCMData(ctx, directClient, configMapNamespace, node)
 	if err != nil {
 		return err
 	}
-	if err = removeServices(svcMgr, services); err != nil {
+	if err = removeServices(latestCMData, versionCMData, svcMgr); err != nil {
 		return err
 	}
+	restartRequired, err := ensureEnvVarsAreRemoved(latestCMData, versionCMData)
+	if err != nil {
+		return err
+	}
+	// rebooting instance to set the environment variables at the process level as expected
+	if restartRequired {
+		// Applying the reboot annotation results in an event picked up by WMCO's node controller to reboot the instance
+		if err = metadata.ApplyRebootAnnotation(ctx, directClient, *node); err != nil {
```

Comment:
Can this lead to a timing issue where
- we apply the reboot annotation onto the node here as part of cleanup
- WICD cleanup returns
- WMCO deletes the node object, before node controller can process reboot annotation event

Reply:
I have thought about this scenario but haven't run into this issue during my testing. I ll take a look again at this.

Reply:
Thanks, concern has been resolved. This was addressed by waiting for reboot annotation removal before deleting node object